### PR TITLE
Implement result event listener

### LINKED ISSUE #74: Implement ResultEventListener
Implement an actor that is able to process specific result events. This issue needs to consider a) that we use akka typed and therefore a corresponding hierarchy in a sealed trait for ResultEventListener needs to be designed and b) that there may be different result events (e.g. final results as GridInputModels or intermediate results) which correspondingly needs a sealed trait hierarchy of processable results. 
Furthermore, there is no need to implement specific writer for the GridInputModel, as we can use existing ones from the PowerSystemDataModel package. 

- [ ] design a result model hierarchy
- [ ] design a result actor 
- [ ] implement the result model hierarchy
- [ ] implement the result actor 
- [ ] write tests for both

### PR DESCRIPTION
**Pull request by johanneshiry**
Resolves #74

### CODE REVIEW

File: src/main/scala/edu/ie3/osmogrid/io/output/PersistenceResultListener.scala [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    Behaviors
+      .receiveMessagePartial[ResultEvent] {
+        case gridResult @ GridResult(grid, replyTo) =>
+          stateData.ctx.pipeToSelf(stateData.sink.handleResult(gridResult)) {
+            case Success(_) => PersistenceSuccessful(stateData.runId)
+            case Failure(exception) =>
+              stateData.ctx.log.error(
+                s"Error during persistence of grid result: $exception"
+              )
+              throw exception
```
Comment (t-ober) [inline src/main/scala/edu/ie3/osmogrid/io/output/PersistenceResultListener.scala:79]: reminder to update error handling and coordinated shutdown

File: src/main/scala/edu/ie3/osmogrid/io/output/PersistenceResultListener.scala [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      with ResultEvent
+
+  // external protocol responses
+  sealed trait ListenerResponse
+  final case class PersistenceSuccessful(runId: UUID)
+      extends ListenerResponse
+      with ResultEvent
+
+  // internal API
+  private sealed trait ResultEvent
```
Comment (t-ober) [inline src/main/scala/edu/ie3/osmogrid/io/output/PersistenceResultListener.scala:46]: reminder to rethink naming

File: src/main/scala/edu/ie3/osmogrid/io/output/ResultCsvSink.scala [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  private val csvFileSink = new CsvFileSink(
+    saveFolderPath,
+    new ProcessorProvider(),
+    if (hierarchic)
+      new HierarchicFileNamingStrategy(
+        "",
+        "",
+        new DefaultDirectoryHierarchy(runId.toString, runId.toString)
+      )
+    else new EntityPersistenceNamingStrategy(),
```
Comment (t-ober) [inline src/main/scala/edu/ie3/osmogrid/io/output/ResultCsvSink.scala:39]:
```suggestion
    if (hierarchic)
      new FileNamingStrategy(
        new EntityPersistenceNamingStrategy(),
        new DefaultDirectoryHierarchy(runId.toString, runId.toString)
      )
    else new FileNamingStrategy(),
```

Comment (t-ober) [reply src/main/scala/edu/ie3/osmogrid/io/output/ResultCsvSink.scala:39]: also take care of imports

**Review by t-ober — DISMISSED** [review]
This is just a quick partial review mainly of the `PersistenceResultListener`and the `ResultCsvSink`as the rest is bound to change according to the adapted `OsmoGridGuardian` changes

**Review by danielfeismann — CHANGES_REQUESTED** [review]
Just some minor things and questions

File: src/main/scala/edu/ie3/osmogrid/io/input/package.scala [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      with InputDataEvent
+  final case class OsmReadFailed(reason: Throwable)
+      extends Response
+      with InputDataEvent
+  final case class RepAssetTypes(assetInformation: AssetInformation)
+      extends Response
+
+  final case class AssetInformation(
+      lineTypes: Seq[LineTypeInput],
+      transformerTypes: Seq[Transformer2WTypeInput]
```
[the remark this answers is not in the document]
Comment (julianhohmann) [reply src/main/scala/edu/ie3/osmogrid/io/input/package.scala:57]: This class has not been changed, it was just moved to a separate file. Maybe we should create a new issue on that?

[the remark this answers is not in the document]
Comment (julianhohmann) [reply src/main/scala/edu/ie3/osmogrid/io/input/package.scala:57]: We have decided together to not consider 3WTransformers in OSMoGrid.

File: src/test/scala/edu/ie3/osmogrid/guardian/run/RunGuardianSpec.scala [hunk 5]
```diff
@@ -141,42 +149,6 @@ class RunGuardianSpec extends ScalaTestWithActorTestKit with UnitSpec {
           s"Received a message, that I don't understand during active run $runId.\n\tMessage: $Run"
         )
       }
-
```
[the remark this answers is not in the document]
Comment (julianhohmann) [reply src/test/scala/edu/ie3/osmogrid/guardian/run/RunGuardianSpec.scala]: Deleted tests are restored.

**Review by danielfeismann — COMMENTED** [review]
Looks good to me. There are some conflicts and Jenkins is atm down. If both is fixed we should be ready to go here

**Review by danielfeismann — APPROVED** [review]
LGTM, thanks for the work done here!

**Pull request merged**